src/lib/utils/settings/abstract_setting.cpp

```diff
@@ -0,0 +1,20 @@
+#include "abstract_setting.hpp"
+
+#include "hyrise.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+AbstractSetting::AbstractSetting(const std::string& init_name) : name(init_name) {}
+
+void AbstractSetting::enroll() {
```

Comment:
register/unregister?

Reply:
I like that better, yes.

Reply:
`register` is a reserved keyword, so something like `register_at_settings_manager()`?

Reply:
Argh, second keyword fail for today. Clunky, but works for me.